lib/semantics/resolve-labels.cc

```diff
@@ -0,0 +1,969 @@
+// Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "resolve-labels.h"
+#include "../common/enum-set.h"
+#include "../parser/message.h"
+#include "../parser/parse-tree-visitor.h"
+#include <cassert>
+#include <cctype>
+#include <cstdarg>
+#include <iostream>
+
+namespace {
+
+using namespace Fortran;
+using namespace parser::literals;
+
+ENUM_CLASS(TargetStatementEnum, Do, Branch, Format)
+using TargetStmtType = common::EnumSet<TargetStatementEnum, 8>;
+
+using IndexList = std::vector<std::pair<parser::CharBlock, parser::CharBlock>>;
+using ScopeProxy = unsigned;
+using LabelStmtInfo = std::tuple<ScopeProxy, parser::CharBlock, TargetStmtType>;
+using TargetStmtMap = std::map<parser::Label, LabelStmtInfo>;
+using SourceStmtList =
+    std::vector<std::tuple<parser::Label, ScopeProxy, parser::CharBlock>>;
+using ErrorHandler = parser::Messages;
+
+const bool isStrictF18{false};  // FIXME - make a command-line option
+
+inline bool HasScope(ScopeProxy scope) { return scope != ScopeProxy{0u}; }
+
+inline bool HasNoScope(ScopeProxy scope) { return !HasScope(scope); }
+
+parser::Message &Report(ErrorHandler &eh, const parser::CharBlock &ip,
+    parser::MessageFormattedText &&msg) {
+  return eh.Say(parser::Message{ip, msg});
```

Comment:
This can just be `return eh.Say(ip, msg);`.

Reply:
sure